### ISSUE TITLE
Feature Request: Enable spellcheck in CodeMirror

### ISSUE THREAD

**Issue by somiaj** [openwebwork/webwork2#1539]
CodeMirror can use the browser's spell checker in the code editor for problems and achievements. The following patch has enabled spell check for me here and I haven't noticed any issues so far.

```diff
diff --git a/htdocs/js/apps/AchievementEditor/achievementeditor.js b/htdocs/js/apps/AchievementEditor/achievementeditor.js
--- a/htdocs/js/apps/AchievementEditor/achievementeditor.js
+++ b/htdocs/js/apps/AchievementEditor/achievementeditor.js
@@ -10,6 +10,8 @@
                                {Tab: function(cm) {cm.execCommand('insertSoftTab')}},
                                highlightSelectionMatches: true,
                                matchBrackets: true,
+                               inputStyle: 'contenteditable',
+                               spellcheck: true,
 
                        });
                cm.setSize("100%", 400);
diff --git a/htdocs/js/apps/PGProblemEditor/pgproblemeditor.js b/htdocs/js/apps/PGProblemEditor/pgproblemeditor.js
--- a/htdocs/js/apps/PGProblemEditor/pgproblemeditor.js
+++ b/htdocs/js/apps/PGProblemEditor/pgproblemeditor.js
@@ -11,6 +11,8 @@
                                {Tab: function(cm) {cm.execCommand('insertSoftTab')}},
                                highlightSelectionMatches: true,
                                matchBrackets: true,
+                               inputStyle: 'contenteditable',
+                               spellcheck: true,
 
                        });
                cm.setSize("100%", 400);
```

**Comment by pstaabp** [comment]
@somiaj Have you tried enabling this?  This seems like a good idea, however since this is mainly for problem editing, I'm guessing it can't handle code from text blocks, and spell checking in code can be annoying.

**Comment by somiaj** [comment]
@pstaabp I am currently using this and it works well. I like the additional check for problem text and just ignore the underlines in rest of the code. I don't find it annoying, but understand if others do. In firefox and chromium (maybe the same for other browsers) the right click in the editor div provides an enable/disable spell check option, so the browser spell checker can be disabled if someone finds it annoying. So once this is enabled in codemirror, the user can turn it on/off in their browser as desired.

**Comment by pstaabp** [comment]
This is also being addressed in #1608
